File: android/app/src/main/java/l2/albitron/scrapyard/cloud/json/Archive.java
package l2.albitron.scrapyard.cloud.json;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

@JsonPropertyOrder({"object", "type", "byte_length"})
public class Archive extends JSONEntity {
    @JsonProperty("object")
    public String object;
    @JsonProperty("type")
    public String type;
    @JsonProperty("byte_length")
    public Long byteLength;
}

File: android/app/src/main/java/l2/albitron/scrapyard/cloud/db/model/Archive.kt
package l2.albitron.scrapyard.cloud.db.model

import com.fasterxml.jackson.annotation.JsonPropertyOrder
import com.fasterxml.jackson.annotation.JsonProperty

@JsonPropertyOrder("type", "content_type")
class Archive : JSONEntity() {
    @JsonProperty("type")
    var type: String? = null

    @JsonProperty("content_type")
    var byteLength: Long? = null
}
